amida-tech/amida-notification-microservice
# Task/orange 1152 sequelize migrations

**Note: It is intentional that this PR merges orange-1152 into orange-1029 because the migrations in 1152 pair with the code changes in 1029.**

#### What's this PR do?

Removes sequelize db.sync() and replaces it with migrations.

#### Related JIRA tickets:

https://jira.amida.com/browse/ORANGE-1152

#### How should this be manually tested?

__Test 1:__

1. Start with a notification service database that is current as of version 1.3.0, and run the migrations on that DB (via `./node_modules/.bin/sequelize db:migrate --to "201902011340-users-uuid-not-null.js"`). Then SQL dump it. (Or, use the file from the email I sent)

2. Create a fresh DB, run the migrations on it. Then SQL dump it.

The only difference between the dump files should be (A) the data in the tables and (B) the presence of the SequelizeMeta table in dump 2.

__Test 2:__

1. Start with a notification service database that is current as of version 1.3.0, _but do not migrate it_. Then SQL dump it. (Or, use the file from the email I sent)

2. Take a notification service database current as of version 1.3.0, then run the migrations on that DB. Then SQL dump it. (Or, use the file from the email I sent)

The only difference between the two dump files should be changes due to these two migrations:
- `201901311643-orange-1029-revoke-device.js`
- `201902011340-users-uuid-not-null.js`

#### Checklist
- [x] I have thoroughly tested this maually. (I didn't give thought to if/how I might write an automated test of it.)
- [x] I have no linter errors
- [x] I have rebased this branch with everything in `feature/ORANGE-1029-revoke-device`
